zeeguu_api/api/dashboard.py

```diff
@@ -0,0 +1,288 @@
+        for link in links:
+            zeeguu.db.session.delete(link)
+        zeeguu.db.session.delete(selected_cohort)
+        zeeguu.db.session.commit()
+        return 'removed'
+    except ValueError:
+        flask.abort(400)
+
+# Takes Teacher id as input and outputs list of all cohort_ids that teacher owns
+@api.route("/get_classes", methods=["GET"])
+@with_session
+def get_classes_by_teacher_id():
+    from zeeguu.model import TeacherCohortMap
+    mappings = TeacherCohortMap.query.filter_by(user_id=flask.g.user.id).all()
+    cohorts = []
+    for m in mappings:
+        info = get_class_info(m.cohort_id)
+        cohorts.append(info)
+    return json.dumps(cohorts)
+
+
+# Takes cohort_id and reuturns dictionary with relevant class variables
+@api.route("/get_class_info/<id>", methods=["GET"])
+@with_session
+def wrapper_to_json_class(id):
+    if(not class_function_checker(id)):
+        flask.abort(401)
+    return jsonify(get_class_info(id))
+
+
+def get_class_info(id):
+    c = Cohort.find(id)
+    class_name = c.class_name
+    inv_code = c.inv_code
+    max_students = c.max_students
+    cur_students = c.cur_students
+    class_language_id = c.class_language_id
+    class_language = Language.query.filter_by(id=class_language_id).one()
+    d = {'id': str(id), 'class_name': class_name, 'inv_code': inv_code, 'max_students': max_students,
+         'cur_students': cur_students, 'class_language_name': class_language.name, 'class_id': id}
+    return d
+
+
+# Takes two inputs (user_id, cohort_id) and links them other in teacher_cohort_map table.
+# url input in format <user_id>/<cohort_id>
+#@api.route("/link_teacher_class/<user_id>/<cohort_id>", methods=["POST"])
+def link_teacher_class(user_id, cohort_id):
+    from zeeguu.model import TeacherCohortMap
+    user = User.find_by_id(user_id)
+    cohort = Cohort.find(cohort_id)
+    zeeguu.db.session.add(TeacherCohortMap(user, cohort))
+    zeeguu.db.session.commit()
+    return 'added teacher_class relationship'
+
+#Checks if the inputted invite code is already in use.
+@api.route("/check_invite_code/<invite_code>", methods=["GET"])
+@with_session
+def check_inv_code(invite_code):
+    c = Cohort.query.filter_by(inv_code=invite_code).first()
+    if c is None:
+        return jsonify(1)
+    return jsonify(0)
+
+
+# creates a class in the data base. Requires form input (inv_code, class_name, class_language_id, max_students, teacher_id)
+@api.route("/add_class", methods=["POST"])
+@with_session
+def add_class():
+    inv_code = request.form.get("inv_code")
+    class_name = request.form.get("class_name")
+    class_language_id = request.form.get("class_language_id")
+    if len(class_language_id) != 2:
+        flask.abort(400)
+    class_language = Language.find_or_create(class_language_id)
+    teacher_id = flask.g.user.id
+    max_students = request.form.get("max_students")
+    if int(max_students) < 1:
+        flask.abort(400)
```

Comment:
we'll need to provide human readable error messages.
for now it's ok but i've added this to the backlog.

https://github.com/RUGSoftEng/2018-ZeeguuDashboard/projects/1#card-9064053

Reply:
also, before starting to work on Human Readable error messages ask @LarsHoldijk  to share with you his Modification Proposal that talks about errors and also maybe have a discussion with him!